examples/stable/test/end_to_end/candid_rpc/candid_backtraces/test/test.ts

```diff
@@ -0,0 +1,139 @@
+import { test } from '@jest/globals';
+import { getCanisterId } from 'azle/_internal/dfx';
+import { expect,runTests } from 'azle/_internal/test';
+import { execSync } from 'child_process';
+
+import { createActor } from './dfx_generated/candid_backtraces';
+
+const canisterName = 'candid_backtraces';
+const testMessage = 'Test trap message for canister backtraces';
+
+function deployCanisterWithEnv(envValue?: string): void {
+    // Clean up first
+    execSync(`dfx canister uninstall-code ${canisterName} || true`, {
+        stdio: 'inherit'
+    });
+
+    // Deploy with or without environment variable
+    const envPrefix = envValue ? `AZLE_CANISTER_BACKTRACES=${envValue} ` : '';
+    execSync(`${envPrefix}dfx deploy ${canisterName}`, {
+        stdio: 'inherit'
+    });
+
+    // Generate types
+    execSync(`dfx generate ${canisterName}`, {
+        stdio: 'inherit'
+    });
+}
+
+function callTrapViaDfx(): string {
+    try {
+        execSync(
+            `dfx canister call ${canisterName} testTrap '("${testMessage}")'`,
+            {
+                stdio: 'pipe'
+            }
+        );
+        throw new Error('Expected dfx call to fail');
+    } catch (error: any) {
+        // The error message contains the full output including backtraces
+        return (
+            error.stderr?.toString() ||
+            error.stdout?.toString() ||
+            error.message
+        );
+    }
+}
+
+runTests(() => {
+    test('normal query works', async () => {
+        // Use current deployment (from pretest)
+        const actor = createActor(getCanisterId(canisterName), {
+            agentOptions: {
+                host: 'http://127.0.0.1:4943',
+                shouldFetchRootKey: true
+            }
+        });
+
+        const result = await actor.testNormalQuery();
+        expect(result).toBe('Normal query response');
+    });
+
+    test('backtraces disabled by default (dfx client)', async () => {
+        // Deploy without AZLE_CANISTER_BACKTRACES
+        deployCanisterWithEnv();
+
+        const dfxOutput = callTrapViaDfx();
+
+        // Should contain the basic error message but no backtrace
+        expect(dfxOutput).toContain(testMessage);
```

Comment:
Make sure to test that the message contains the canister method name that we called